perfmetrics/scripts/ls_metrics/listing_benchmark_test.py

```diff
@@ -0,0 +1,519 @@
+                                        'num_folders': 2,
+                                        'folders': [
+                                            {
+                                                'name': 'subsubdir1',
+                                                'num_files': 2,
+                                                'file_name_prefix': 'file',
+                                                'file_size': '1kb'
+                                            },
+                                            {
+                                                'name': 'subsubdir2'
+                                            }
+                                        ]
+                            },
+                            {
+                                'name': 'subdir2',
+                                'num_files': 1,
+                                'file_name_prefix': 'file',
+                                        'file_size': '1kb'
+                            },
+                            {
+                                'name': 'subdir3',
+                                'num_files': 1,
+                                'file_name_prefix': 'file',
+                                        'file_size': '1kb',
+                                        'num_folders': 1,
+                                        'folders': [
+                                            {
+                                                'name': 'subsubdir1',
+                                                'num_files': 1,
+                                                'file_name_prefix': 'file',
+                                                'file_size': '1kb'
+                                            }
+                                        ]
+                            }
+                        ]
+            },
+        {
+                'name': '2KB_3files_1subdir',
+                'num_files': 3,
+                'file_name_prefix': 'file',
+                        'file_size': '2kb',
+                        'num_folders': 1,
+                        'folders': [
+                            {
+                                'name': 'subdir1'
+                            }
+                        ]
+                },
+        {
+                'name': '1KB_1files_0subdir',
+                'num_files': 1,
+                'file_size': '1kb',
+                'file_name_prefix': 'file'
+                }
+    ]
+}
+
+METRICS1 = [1.234, 0.995, 0.121, 0.222, 0.01709]
+METRICS2 = [90.45, 1.95, 0.334, 7.090, 0.001]
+METRICS3 = [100, 7, 6, 51, 21]
+
+DIRECTORY_STRUCTURE1 = ParseDict(
+    DIRECTORY_STRUCTURE1, directory_proto.Directory())
+DIRECTORY_STRUCTURE2 = ParseDict(
+    DIRECTORY_STRUCTURE2, directory_proto.Directory())
+DIRECTORY_STRUCTURE3 = ParseDict(
+    DIRECTORY_STRUCTURE3, directory_proto.Directory())
+
+
+class ListingBenchmarkTest(unittest.TestCase):
+
+    def test_parse_results_type1(self):
```

Comment:
Can we have better names here? results_single_level_dir and results_multi_level_dir etc.?

Reply:
changed.